dromedary.py

```diff
@@ -163,6 +180,143 @@ def print_output(out: subprocess.CompletedProcess) -> None:
     print(std_out + err_out)
 
 
+###############################################################################
+def run_cmd_output(cmd_args: List[Any], cmd_env: Optional[Dict[str, str]]) -> str:
+    """Run the given command in a shell.
+
+    Prints the output of the process in "real time".
+
+    Args:
+        cmd_args (List[Any]): The command and it's arguments to run.
+        cmd_env (Optional[Dict[str, str]]): The environment to run the command
+                                            in.
+
+    Returns:
+        str: The accumulated stdout and stderr output of the process.
+    """
+    std_out: str = ""
+    with subprocess.Popen(
+        " ".join(map(lambda e: f"'{str(e)}'", cmd_args)),
+        shell=True,
+        env=cmd_env,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+    ) as p:  # nosec
+        while p.poll() is None:
+            std_str = p.stdout.read1().decode("utf-8")
+            std_out += std_str
+            print(std_str, end="", flush=True, file=sys.stdout)
+
+        if p.poll() != 0:
+            print(f"Error: command '{p.args}' returned '{p.poll()}'")
+            return error_exit(6)
+
+    return std_out
+
+
+###############################################################################
+def generate_switch(config_file: str) -> Tuple[str, Dict[str, Any]]:
+    """Generate an Opam switch from the configuration in the given JSON file
+    `config_file`.
+
+    Args:
+        config_file (str): The path to the JSON config file to use.
+
+    Returns:
+        Tuple[str, Dict[str, Any]]: The name of the generated Opam switch and
+                                    the content of the JSON file `config_file`.
+    """
+    print(f"Using config file '{config_file}'")
+    switch_config = read_json(config_file)
+
+    gen_switch_name = switch_config.get("name")
+    if gen_switch_name is None:
+        gen_switch_name = DEFAULT_OPAM_SWITCH_NAME
+
+    if gen_switch_name.startswith("."):
```

Comment:
Be careful of `..`

Reply:
I've chosen `.`, so that it works with `./`, `../` and hidden directories like `.NAME`, which is what Opam interprets as a relative path (I guess they use the same test).
 
I guess I'll add a comment ;)

Reply:
I was not aware of that :) TIL !